content/tracing/languages/java.md

```diff
@@ -0,0 +1,274 @@
+
+```java
+import datadog.opentracing.DDTracer;
+import io.opentracing.util.GlobalTracer;
+import datadog.trace.api.sampling.AllSampler;
+
+public class Application {
+
+    public static void main(String[] args) {
+
+        // Initialize the tracer from environment variables or system properties
+        Tracer tracer = new DDTracer();
+        GlobalTracer.register(tracer);
+
+        // OR from the API
+        Writer writer = new DDAgentWriter();
+        Sampler sampler = new AllSampler();
+        Tracer tracer = new DDTracer(writer, sampler);
+        GlobalTracer.register(tracer);
+
+        // ...
+    }
+}
+```
+
+Notice the above examples only use the OpenTracing classes. Please reference the [OpenTracing API](https://github.com/opentracing/opentracing-java) for more details and information.
+
+## Sampling / distributed tracing
+
+Priority sampling ensures that distributed traces are complete by assigning and propagating a priority value along each trace.
+
+Current Priority Values (more may be added in the future):
+
+|Sampling Value | Effect                                                                                                     |
+|---------------|:----------------------------------------------------------------------------------------------------------:|
+|SAMPLER_DROP   | The sampler automatically decided to not keep the trace. The Agent will drop it.                           |
+|SAMPLER_KEEP   | The sampler automatically decided to keep the trace. The Agent will keep it. Might be sampled server-side. |
+|USER_DROP      | The user asked to not keep the trace. The Agent will drop it.                                              |
+|USER_KEEP      | The user asked to keep the trace. The Agent will keep it. The server will keep it too.                     |
+
+
+Priority sampling is disabled by default. To enable it, configure the `priority.sampling` flag to `true`. [How to configure the client?](# Configuration).
+
+Once enabled, the sampler automatically assign a priority to traces, depending on their service and volume.
+
+You can also set this priority manually to either drop a non-interesting trace or to keep an important one.
+```java
+import datadog.opentracing.DDSpan;
+import datadog.trace.api.Trace;
+import datadog.trace.common.sampling.PrioritySampling;
+import io.opentracing.util.GlobalTracer;
+
+public class MyClass {
+  @Trace
+  public static void myMethod() {
+    // grab the active span out of the traced method
+    DDSpan ddspan = (DDSpan) GlobalTracer.get().activeSpan();
+    // ask the sampler to keep the current trace
+    ddspan.setSamplingPriority(PrioritySampling.USER_KEEP);
+    // method impl follows
+  }
+}
+```
+
+## Integrations
+
+### Web Frameworks
+
+`dd-java-agent` includes support for automatically tracing the following web frameworks.
+
+| Server | Versions |
```

Comment:
Column alignment for `Versions` in the [published version](https://docs-staging.datadoghq.com/gus/apm-java/tracing/languages/java/#networking-frameworks) looks funny...

Reply:
👍